dll/win32/shell32/CCopyMoveToMenu.cpp

```diff
@@ -436,3 +436,145 @@ CCopyMoveToMenu::GetSite(REFIID riid, void **ppvSite)
 
     return m_pSite->QueryInterface(riid, ppvSite);
 }
+
+static DWORD
+SetClipboard(UINT cf, const void* data, SIZE_T size)
+{
+    BOOL succ = FALSE;
+    HGLOBAL handle = GlobalAlloc(GMEM_MOVEABLE | GMEM_SHARE, size);
+    if (handle)
+    {
+        LPVOID clipdata = GlobalLock(handle);
+        if (clipdata)
+        {
+            CopyMemory(clipdata, data, size);
+            GlobalUnlock(handle);
+            if (OpenClipboard(NULL))
+            {
+                EmptyClipboard();
+                succ = SetClipboardData(cf, handle) != NULL;
+                CloseClipboard();
+            }
+        }
+        if (!succ)
+        {
+            GlobalFree(handle);
+        }
+    }
+    return succ ? ERROR_SUCCESS : GetLastError();
```

Comment:
is that call to GetLastError() inevitable for proper compatibility? It's such a poor 80ies-like API.

Reply:
In theory you could just force return E_FAIL. Who knows if the ultimate caller will care about GlobalAlloc out of memory vs OpenClipboard failure.